### ISSUE TITLE
MunkiAdmin 'hangs' when running makecatalogs

### ISSUE THREAD

**Issue by bochoven** [hjuutilainen/munkiadmin#118]
When pressing the 'Save' button, MA shows a dialog that says: 'Status: Running makecatalogs' which never ends. Pressing 'Cancel' does not make the dialog go away and the only left to do is to force quit Munkiadmin.

With Log level set to Verbose, I get the following log entry:

```
2016/03/26 09:14:02:753  MunkiOperation:makecatalogs
2016/03/26 09:14:02:753  Running /usr/local/munki/makecatalogs with arguments: (
    "--force",
    "/Users/Shared/munki"
)
2016/03/26 09:14:03:017  startOperationTimer
```

**Comment by hjuutilainen** [comment]
That's a weird one, took me a while to get to the bottom of (or at least get a theory) of what's happening. I think there is a deadlock issue in the MunkiAdmin's makecatalogs operation visible only if there's a lot of stderr output.

Could you try this custom build to see if it helps:
https://www.dropbox.com/s/wlon930g0k3tbi9/MunkiAdmin-1.4.1-11-gf7e0caf.dmg?dl=0

In that build I just changed the order in which MunkiAdmin reads stderr and stdout from makecatalogs. A quick and dirty fix without fully understanding what's going on. Needs more investigation...

**Comment by bochoven** [comment]
The new build is completing succesfully. Good Job!

**Comment by hjuutilainen** [comment]
This is now fixed in version 1.4.2.
